mne/stats/regression.py

```diff
@@ -134,3 +136,193 @@ def _fit_lm(data, design_matrix, names):
+        and the observation matrix Y, and returns the coefficient matrix b;
+        or a string (for now, only 'pinv'), in which case the solver used
+        is dot(scipy.linalg.pinv(dot(X.T, X)), dot(X.T, Y.T)).T.
+
+    Returns
+    -------
+    ev_dict : dictionary
+        A dictionary where the keys correspond to conditions and the values are
+        Evoked objects with the rE[R/F]Ps. These can be used exactly like any
+        other Evoked object, including e.g. plotting or statistics.
+    """
+
+    if isinstance(solver, string_types):
+        if solver == 'pinv':
+            fast_dot = _get_fast_dot()
+
+            # inv is slightly (~10%) faster, but pinv seemingly more stable
+            def solver(X, Y):
+                return fast_dot(linalg.pinv(fast_dot(X.T, X)),
+                                fast_dot(X.T, Y.T)).T
+        else:
+            raise ValueError("No such solver: {0}".format(solver))
+
+    # prepare raw and events
+    if picks is None:
+        picks = pick_types(raw.info, meg=True, eeg=True, ref_meg=True)
+    info = pick_info(raw.info, picks, copy=True)
+    info["sfreq"] /= decim
+    data, times = raw[picks, ::decim]
+    events = events.copy()
+    events[:, 0] -= raw.first_samp
+    events[:, 0] /= decim
+
+    conds = list(event_id)
+    if covariates is not None:
+        conds += list(covariates)
+
+    # time windows (per event type)
+    if isinstance(tmin, (float, int)):
+        tmin = dict((cond, int(tmin * info["sfreq"])) for cond in conds)
+    else:
+        tmin = dict((cond, int(tmin.get(cond, -.1) * info["sfreq"]))
+                    for cond in conds)
+    if isinstance(tmax, (float, int)):
+        tmax = dict((cond, int((tmax * info["sfreq"]) + 1.)) for cond in conds)
+    else:
+        tmax = dict((cond, int((tmax.get(cond, 1.) * info["sfreq"]) + 1))
+                    for cond in conds)
+
+    # Construct predictor matrix
+    # We do this by creating one array per event type, shape lags * samples
+    # (where lags depends on tmin/tmax and can be different for different
+    # event types). Columns correspond to predictors, predictors correspond to
+    # time lags. Thus, each array is mostly sparse, with one diagonal of 1s
+    # per event (for binary predictors).
+
+    # This should probably be improved (including making it more robust to
+    # high frequency data) by operating on sparse matrices. As-is, high
+    # sampling rates plus long windows blow up the system due to the inversion
+    # of massive matrices.
+    # Furthermore, assigning to a preallocated array would be faster.
+
+    cond_length = dict()
+    pred_arrays = list()
+    for cond in conds:
+        # create the first row and column to be later used by toeplitz to build
+        # the full predictor matrix
+        tmin_, tmax_ = tmin[cond], tmax[cond]
+        n_lags = int(tmax_ - tmin_)
+        samples = np.zeros(len(times), dtype=np.float64)
+        lags = np.zeros(n_lags, dtype=np.float64)
```

Comment:
It's the default vOv
Anything wrong with it?

Eventually this should become sparse.